infrastructure/containerd/image_service_integration_test.go

```diff
@@ -0,0 +1,215 @@
+package containerd_test
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"testing"
+
+	ctr "github.com/containerd/containerd"
+	"github.com/containerd/containerd/namespaces"
+	"github.com/containerd/containerd/snapshots"
+	. "github.com/onsi/gomega"
+
+	"github.com/weaveworks/flintlock/core/models"
+	"github.com/weaveworks/flintlock/core/ports"
+	"github.com/weaveworks/flintlock/infrastructure/containerd"
+)
+
+const (
+	testImageVolume    = "docker.io/library/alpine:3.14.1"
+	testImageKernel    = "docker.io/linuxkit/kernel:5.4.129"
+	testSnapshotter    = "devmapper"
+	testOwnerNamespace = "int_ns"
+	testOwnerUsageID   = "vol1"
+	testOwnerName      = "imageservice-get-test"
+	testContainerdNS   = "flintlock_test_ctr"
+)
+
+func TestImageService_Integration(t *testing.T) {
+	if !runContainerDTests() {
+		t.Skip("skipping containerd image service integration test")
+	}
+
+	RegisterTestingT(t)
+
+	client, ctx := testCreateClient(t)
+	namespaceCtx := namespaces.WithNamespace(ctx, testContainerdNS)
+
+	imageSvc := containerd.NewImageServiceWithClient(&containerd.Config{
+		SnapshotterKernel: testSnapshotter,
+		SnapshotterVolume: testSnapshotter,
+		Namespace:         testContainerdNS,
+	}, client)
+
+	inputGetAndMount := &ports.ImageMountSpec{
+		ImageName:    getTestVolumeImage(),
+		Owner:        fmt.Sprintf("%s/%s", testOwnerNamespace, testOwnerName),
+		OwnerUsageID: testOwnerUsageID,
+		Use:          models.ImageUseVolume,
+	}
+	inputGet := &ports.ImageSpec{
+		ImageName: inputGetAndMount.ImageName,
+		Owner:     inputGetAndMount.Owner,
+	}
+	expectedSnapshotName := fmt.Sprintf(
+		"flintlock/%s/%s/%s",
+		testOwnerNamespace,
+		testOwnerName,
+		testOwnerUsageID,
+	)
+	expectedLeaseName := fmt.Sprintf("flintlock/%s/%s", testOwnerNamespace, testOwnerName)
+
+	defer func() {
+		// Make sure it's deleted.
+		client.ImageService().Delete(namespaceCtx, testImageKernel)
+		client.ImageService().Delete(namespaceCtx, testImageVolume)
```

Comment:
the volume image used on line 46 might be different than the one being cleaned up, did you mean to use `getTestVolumeImage()` here (and on a few other lines which have `testImageVolume`)?